src/Program.ts

```diff
@@ -705,42 +779,73 @@ export class Program {
         return file.getHover(position);
     }
 
-    public async getSignatureHelp(callSitePathAbsolute: string, callableName: string) {
-        const results = [] as SignatureInformation[];
-
-        callableName = callableName.toLowerCase();
-
-        //find the file
-        let file = this.getFile(callSitePathAbsolute);
-        if (!file) {
-            return results;
+    public async getSignatureHelp(filepath: string, position: Position): Promise<SignatureInfoObj[]> {
+        let file: BrsFile = this.getFile(filepath);
+        if (!file || !isBrsFile(file)) {
+            return [];
         }
 
-        const scopes = this.getScopesForFile(file);
-        for (const scope of scopes) {
-            const files = scope.getOwnFiles();
-            for (const file of files) {
-                if (isXmlFile(file)) {
-                    continue;
-                }
-
-                await file.isReady();
+        await file.isReady();
 
-                const statementHandler = (statement: FunctionStatement | ClassMethodStatement) => {
-                    if (statement.getName(file.getParseMode()).toLowerCase() === callableName) {
-                        results.push(file.getSignatureHelp(statement));
+        const results = new Map<string, SignatureInfoObj>();
+        const statementHandler = (expr: CallExpression | CallfuncExpression | NewExpression) => {
+            if (util.rangeContains(expr.range, position)) {
+                if (isCallExpression(expr)) {
```

Comment:
So little of this statementhandler is shared between the different expressions, there's not much point in sharing the same function for all three. You should just add inline handlers in the `walk` function below instead, one for each of the specific expression types.

Reply:
indeed, this one started out as a good idea and ended like this - good catch. thanks.